File: avaliacao_tcs/src/java/DAO/DAOCliente.java
package DAO;
import MODEL.Cliente;
import java.util.ArrayList;
import java.sql.ResultSet;

/**
 *
 * @author ITALO PIOVAN
 */
public class DAOCliente 
{
    private static DAOCliente instance;
    private Conexao conn = new Conexao();
    
    private DAOCliente() {}
    
    public static synchronized DAOCliente getInstance() 
    {
        if (instance == null)
            instance = new DAOCliente();
        return instance;
    }
    
    public boolean insertCliente(Cliente cliente)
    {
        boolean success = false;
        String sql = "INSERT INTO tb_cliente (nome_cliente, email_cliente, telefone_cliente) VALUES ('@a', '@b', '@c')";
        
        sql = sql.replace("@a", cliente.getNome_cliente());
        sql = sql.replace("@b", cliente.getEmail_cliente());
        sql = sql.replace("@c", cliente.getTelefone_cliente());
        
        conn.openConnection();
        success = conn.manipulate(sql);
        conn.closeConnection();
        return success;
    }
    
    public boolean updateCliente(Cliente cliente)
    {
        boolean success = false;
        String sql = "UPDATE tb_cliente SET nome_cliente = '@a', email_cliente = '@b', telefone_cliente = '@c' WHERE id_cliente = @d";
        
        sql = sql.replace("@a", cliente.getNome_cliente());
        sql = sql.replace("@b", cliente.getEmail_cliente());
        sql = sql.replace("@c", cliente.getTelefone_cliente());
        sql = sql.replace("@d", Integer.toString(cliente.getId_cliente()));
        
        conn.openConnection();
        success = conn.manipulate(sql);
        conn.closeConnection();
        return success;
    }
    
    public boolean deleteCliente(int id)
    {
        boolean success = false;
        String sql = "DELETE FROM tb_cliente WHERE id_cliente = @a";
   
        sql = sql.replace("@a", Integer.toString(id));
        
        conn.openConnection();
        success = conn.manipulate(sql);
        conn.closeConnection();
        return success;
    }
    
    public ArrayList<Cliente> getClientes()
    {
        ArrayList<Cliente> clientes = new ArrayList<Cliente>();
        try{
            String sql = "SELECT * FROM tb_cliente";
            
            conn.openConnection();
            ResultSet rs = conn.query(sql);
            
            if(rs != null)
                while(rs.next())
                    clientes.add(new Cliente(rs.getInt("id_cliente"), rs.getString("nome_cliente"), 
                                             rs.getString("email_cliente"), rs.getString("telefone_cliente")));
            
            conn.closeConnection();   
        }catch(Exception e){throw new RuntimeException("Erro ao listar clientes" + e);}
        return clientes;
    }
    
    public ArrayList<Cliente> getClientesFilter(String nome)
    {
        ArrayList<Cliente> clientes = new ArrayList<Cliente>();
        try{
            String sql = "SELECT * FROM tb_cliente WHERE nome_cliente LIKE '%" +nome+ "%'";
            
            conn.openConnection();
            ResultSet rs = conn.query(sql);
            
            if(rs != null)
                while(rs.next())
                    clientes.add(new Cliente(rs.getInt("id_cliente"), rs.getString("nome_cliente"), 
                                             rs.getString("email_cliente"), rs.getString("telefone_cliente")));
            
            conn.closeConnection();
        }catch(Exception e){throw new RuntimeException("Erro ao listar clientes" + e);}
        return clientes;
    }
    
    public Cliente getClienteById(int id)
    {
        Cliente cliente = null;
        try{
            String sql = "SELECT * FROM tb_cliente WHERE id_cliente = "+id;
            
            conn.openConnection();
            ResultSet rs = conn.query(sql);
            
            if(rs != null && rs.next())
                cliente = new Cliente(rs.getInt("id_cliente"), rs.getString("nome_cliente"),rs.getString("email_cliente"), rs.getString("telefone_cliente"));
            
            conn.closeConnection();
        }catch(Exception e){throw new RuntimeException("Erro ao listar cliente pelo id" + e);}
        return cliente;
    }
}

File: CRUD C Sharp/DAO/DAOCliente.cs
using CRUD_C_Sharp.Models;
using System;
using System.Collections.Generic;
using System.Data;

namespace CRUD_C_Sharp.DAO
{
    public class DAOCliente
    {
        MySQLPersistence _bd = new MySQLPersistence();
        private static DAOCliente instance;
        private DAOCliente() { }
        public static DAOCliente getInstance()
        {
            if (instance == null)
                instance = new DAOCliente();
            return instance;
        }

        public bool insertCliente(Cliente cliente)
        {
            bool success = false;
            try
            {
                string sql = "INSERT INTO tb_cliente (nome_cliente, email_cliente, telefone_cliente) VALUES (@a, @b, @c)";

                _bd.clearParameter();
                _bd.addParameter("@a", cliente.Nome);
                _bd.addParameter("@b", cliente.Email);
                _bd.addParameter("@c", cliente.Telefone);

                _bd.openConnection();

                if (_bd.manipulate(sql) >= 1)
                    success = true;

                _bd.closeConnection();
            }
            catch (Exception ex){ throw new Exception("Erro" + ex); }
            return success;
        }
        public bool updateCliente(Cliente cliente)
        {
            bool success = false;
            try
            {
                string sql = "UPDATE tb_cliente SET nome_cliente = @a, email_cliente = @b, telefone_cliente = @c WHERE id_cliente = @d";

                _bd.clearParameter();
                _bd.addParameter("@a", cliente.Nome);
                _bd.addParameter("@b", cliente.Email);
                _bd.addParameter("@c", cliente.Telefone);
                _bd.addParameter("@d", cliente.Id);

                _bd.openConnection();

                if (_bd.manipulate(sql) >= 1)
                    success = true;

                _bd.closeConnection();
            }
            catch (Exception ex) { throw new Exception("Erro" + ex); }
            return success;
        }

        public bool deleteCliente(int id)
        {
            bool success = false;
            try
            {
                string sql = "DELETE FROM tb_cliente WHERE id_cliente = @a";

                _bd.clearParameter();
                _bd.addParameter("@a", id);
               
                _bd.openConnection();

                if (_bd.manipulate(sql) >= 1)
                    success = true;

                _bd.closeConnection();
            }
            catch (Exception ex) { throw new Exception("Erro" + ex); }
            return success;
        }

        public IEnumerable<Cliente> getClientes()
        {
            List<Cliente> clientes = new List<Cliente>();
            try
            {
                string sql = "SELECT * FROM tb_cliente";
                _bd.openConnection();
                DataTable dt = _bd.executeQuery(sql);
                _bd.closeConnection();
                foreach (DataRow row in dt.Rows)
                {
                    var cli = new Cliente()
                    {
                        Id = Convert.ToInt32(row["id_cliente"]),
                        Nome = row["nome_cliente"].ToString(),
                        Email = row["email_cliente"].ToString(),
                        Telefone = row["telefone_cliente"].ToString()
                    };
                    clientes.Add(cli);
                }

            }
            catch (Exception ex) { throw new Exception("Erro" + ex); }
            return clientes;
        }
        public IEnumerable<Cliente> getClientesFilter(string nome)
        {
            List<Cliente> clientes = new List<Cliente>();
            try
            {
                string sql = "SELECT * FROM tb_cliente WHERE nome_cliente LIKE '%"+nome+"%'";
                _bd.openConnection();
                DataTable dt = _bd.executeQuery(sql);
                _bd.closeConnection();
                foreach (DataRow row in dt.Rows)
                {
                    var cli = new Cliente()
                    {
                        Id = Convert.ToInt32(row["id_cliente"]),
                        Nome = row["nome_cliente"].ToString(),
                        Email = row["email_cliente"].ToString(),
                        Telefone = row["telefone_cliente"].ToString()
                    };
                    clientes.Add(cli);
                }
            }
            catch (Exception ex) { throw new Exception("Erro" + ex); }
            return clientes;
        }

        public IEnumerable<Cliente> getClientesById(int id)
        {
            List<Cliente> clientes = new List<Cliente>();
            try
            {
                string sql = "SELECT * FROM tb_cliente WHERE id_cliente = " + id;
                _bd.openConnection();
                DataTable dt = _bd.executeQuery(sql);
                _bd.closeConnection();
                foreach (DataRow row in dt.Rows)
                {
                    var cli = new Cliente()
                    {
                        Id = Convert.ToInt32(row["id_cliente"]),
                        Nome = row["nome_cliente"].ToString(),
                        Email = row["email_cliente"].ToString(),
                        Telefone = row["telefone_cliente"].ToString()
                    };
                    clientes.Add(cli);
                }
            }
            catch (Exception ex) { throw new Exception("Erro" + ex); }
            return clientes;
        }
    }
}
